Lib/sqlite3/__main__.py

```diff
@@ -35,57 +39,65 @@ def runpy(self, source, filename="<input>", symbol="single"):
 
     def printhelp(self, ignored):
         print("Enter SQL code and press enter.")
+        return None
 
     def runsource(self, source, filename="<input>", symbol="single"):
         keywords = {
-            "version": lambda x: print(f"{sqlite3.sqlite_version}"),
-            "help": self.printhelp,
-            "quit()": self.runpy,
-            "quit": self.runpy,
+            ".version": lambda x: print(f"{sqlite3.sqlite_version}"),
+            ".help": self.printhelp,
+            ".quit": lambda x: sys.exit(0),
         }
         return keywords.get(source, self.runsql)(source)
 
 
-def dump_version():
-    print(f"SQLite version {sqlite3.sqlite_version}")
-
-
 def main():
     parser = ArgumentParser(
         description="Python sqlite3 REPL",
         prog="python -m sqlite3",
     )
     parser.add_argument(
-        "-f", "--filename",
-        type=str, dest="database", action="store", default=":memory:",
-        help="Database to open (defaults to ':memory:')",
+        "filename", type=str, default=":memory:", nargs="?",
+        help=(
+            "SQLite database to open (defaults to ':memory:'). "
+            "A new database is created if the file does not previously exist."
+        ),
+    )
+    parser.add_argument(
+        "sql", type=str, nargs="?",
+        help=(
+            "An SQL query to execute. "
+            "Any returned rows are printed to stdout."
+        ),
     )
     parser.add_argument(
-        "-v", "--version",
-        dest="show_version", action="store_true", default=False,
+        "-v", "--version", action="version",
+        version=f"SQLite version {sqlite3.sqlite_version}",
         help="Print underlying SQLite library version",
     )
     args = parser.parse_args()
-    if args.show_version:
-        return dump_version()
 
-    if args.database == ":memory:":
+    if args.filename == ":memory:":
         db_name = "a transient in-memory database"
     else:
-        db_name = f"'{args.database}'"
+        db_name = repr(args.filename)
 
     banner = dedent(f"""
         sqlite3 shell, running on SQLite version {sqlite3.sqlite_version}
         Connected to {db_name}
 
         Each command will be run using execute() on the cursor.
-        Type "help" for more information; type "quit" or CTRL-D to quit.
+        Type ".help" for more information; type ".quit" or CTRL-D to quit.
     """).strip()
     sys.ps1 = "sqlite> "
     sys.ps2 = "    ... "
 
-    console = SqliteInteractiveConsole(args.database)
-    console.interact(banner, exitmsg="")
+    con = sqlite3.connect(args.filename, isolation_level=None)
+    if args.sql:
+        execute(con, args.sql)
+    else:
+        console = SqliteInteractiveConsole(con)
+        console.interact(banner, exitmsg="")
+    con.close()
```

Comment:
try/finally to guarantee closing on exit.

Reply:
Good point